domain/application/service/application.go

```diff
@@ -713,7 +715,29 @@ func (s *Service) UpdateCAASUnit(ctx context.Context, unitName coreunit.Name, pa
 		return internalerrors.Errorf("application %q is not alive%w", appName, errors.Hide(applicationerrors.ApplicationNotAlive))
 	}
 
-	if err := s.st.UpdateCAASUnit(ctx, unitName, params); err != nil {
+	agentStatus, err := encodeUnitAgentStatus(params.AgentStatus)
+	if err != nil {
+		return internalerrors.Errorf("encoding agent status: %w", err)
+	}
+	workloadStatus, err := encodeUnitWorkloadStatus(params.WorkloadStatus)
+	if err != nil {
+		return internalerrors.Errorf("encoding workload status: %w", err)
+	}
+	cloudContainerStatus, err := encodeCloudContainerStatus(params.CloudContainerStatus)
+	if err != nil {
+		return internalerrors.Errorf("encoding cloud container status: %w", err)
+	}
+
+	cassUnitUpdate := application.UpdateCAASUnitParams{
```

Comment:
This type `application.UpdateCAASUnitParams` is now only a param for a state method, and constructed in the service to call said state method.

Thank you for bifurcating this params struct, this was something I missed when messing with this method myself.

But this remaining param struct should probably be moved back to `application/state/params.go`

Reply:
It's used between state and service, so you can't.